validate.go

```diff
@@ -94,17 +95,24 @@ func putValidate(r *http.Request) (interface{}, nap.Status) {
 
 	config := bytes.Replace(body, []byte("\r"), []byte{}, -1)
 
-	_, report, err := ignConfig.Parse(config)
+	_, rpt, err := ignConfig.Parse(config)
 	switch err {
 	case ignConfig.ErrCloudConfig, ignConfig.ErrEmpty, ignConfig.ErrScript:
-		report, err := validate.Validate(config)
+		rpt, err := validate.Validate(config)
 		if err != nil {
 			return nil, nap.InternalError{err.Error()}
 		}
-		return report.Entries(), nap.OK{}
+		return rpt.Entries(), nap.OK{}
+	case ignConfig.ErrUnknownVersion:
+		return report.Report{
+			Entries: []report.Entry{{
+				Kind:    report.EntryError,
+				Message: "Failed to parse config. Is this a valid Ignition Config, Cloud-Config, or script?",
+			}},
+		}, nap.OK{}
 	default:
-		report.Sort()
-		return report.Entries, nap.OK{}
+		rpt.Sort()
```

Comment:
Actually, looking at this further. It should be okay to catch this in the default case. My code ended up looking like the following:

```go
        case nil:
                rpt.Sort()
                return rpt.Entries, nap.OK{}
        default:
                rpt.Sort()
                return rpt.Entries, nap.OK{}
```